src/util/psbt/map/global.rs

```diff
@@ -92,6 +106,34 @@ impl Map for Global {
             },
         });
 
+        for (xpub, (fingerprint, derivation)) in &self.xpub {
+            rv.push(raw::Pair {
+                key: raw::Key {
+                    type_value: PSBT_GLOBAL_KEY_XPUB,
+                    key: xpub.encode().to_vec(),
+                },
+                value: {
+                    let mut ret = Vec::with_capacity(4 + derivation.len() * 4);
+                    ret[0..4].copy_from_slice(fingerprint.as_bytes());
```

Comment:
This will cause runtime error. `ret` is a vec with length 0, so you can't use `vec[0..4]`

Reply:
Fixed